### ISSUE TITLE
[Bug]: Error message is not shown in git connect

### ISSUE THREAD

**Issue by AnaghHegde** [appsmithorg/appsmith#13179]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Error toast is not shown for few cases in git connect when the key does not have write access. 

### Steps To Reproduce

1. Create a repo in bitbucket
2. Generate the key for connecting the app from Appsmith
3. Add the key in bitbucket repo settings
4. Notice that no error is shown to the user
5. Open the browser console and observe that there are errors thrown from browser due to missing permission for the keys
![Uploading Screenshot 2022-04-21 at 2.54.08 PM.png…]()

### Version

Cloud

**Comment by vuiets** [comment]
@AnaghHegde @RakshaKShetty does this happen only for Bitbuckets or for all providers?

**Comment by AnaghHegde** [comment]
@vuiets I am not able to reproduce this now. I was able to reproduce this when i was in debug session with a user.

**Comment by IAmAnubhavSaini** [comment]
@AnaghHegde was the user trying to add access key to repo.

<img width="1259" alt="Screenshot 2022-05-02 at 2 52 01 PM" src="https://user-images.githubusercontent.com/1573771/166212889-79a66044-6be5-488c-82c2-0c08bb8a83c2.png">

That is not supported for bitbucket, users need to add key to account.

Personal Settings -> SSH keys

<img width="1237" alt="Screenshot 2022-05-02 at 2 53 05 PM" src="https://user-images.githubusercontent.com/1573771/166213070-41c68ca9-3d78-4c99-be94-93f164fd7d34.png">

**Comment by AnaghHegde** [comment]
@IAmAnubhavSaini Yes i agree. But the error message was not showing up in the git sync modal.

**Comment by IAmAnubhavSaini** [comment]
Really need to be able to reproduce the bug. 

Do you have all the steps, because it feels like one is missing, after 3 above?

---

The message comes up when we try to change to branch or push commit. Let me know. Code: https://github.com/appsmithorg/appsmith/blob/release/app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java#L946-L947

---

I got this message when I added SSH key to repo settings and tried to connect.

<img width="1590" alt="Screenshot 2022-05-02 at 2 56 23 PM" src="https://user-images.githubusercontent.com/1573771/166214230-fc9433f5-71dd-410f-8147-6bcfddc5ac55.png">

**Comment by vuiets** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 25 |
| Effort (months)  | 0.5 |

If this is for Bitbuckets in particular, reach is 25

**Comment by IAmAnubhavSaini** [comment]
This has been fixed.

![Screenshot 2022-06-17 at 1 58 41 PM](https://user-images.githubusercontent.com/1573771/174259282-9a5a8b9a-bc64-47ab-a4b5-e9e36bb6b1a2.png)
